p_sets/2/vigenere.rb

```diff
@@ -0,0 +1,61 @@
+# Accept a single command-line argument: a keyword, k
+# composed entirely of alphabetical characters.
+# If your program is executed with 0, or more than 1 CLA,
+# or if it contains any non-alphabetical chars exit and return 1.
+# Treat A and a as 0, B and b as 1, and Z and z as 25.
+# Only apply cipher to letters. Preserve the case.
+# All other characters must be outputted unchanged.
+# If your code is about to apply the jth character of k to
+# a non-alphabetical character, you must wait to apply it
+# to the next alphabetical character. Do not advance
+# to the next character in k.
+require_relative 'encryption'
+class Vigenere
+  include Encryption
+
+  def make_key(string)
+    key = []
+    string.upcase.each_char do |letter|
+      key << letter.ord - ASCII_LETTERS[:capitals].min
+    end
```

Comment:
Another case for a `map`. It's not a very C++ concept, so it might not be covered in the course.